classes/music_album.rb

```diff
@@ -0,0 +1,14 @@
+require_relative 'item'
+
+class MusicAlbum < Item
+  attr_accessor :on_sportify, :publish_date
+
+  def initialize(on_sportify, publish_date)
+    super(publish_date)
+    @on_sportify = on_sportify
+  end
+
+  def can_be_archived?()
+    super || @on_sportify == true
+  end
+end
```

Comment:
Just after line 13 in the item.rb file , you can add the add_genre method which you will call in the add_item method. It should look something like this. 

```
def add_genre(genre)
    @genre = genre
    genre.add_item(self) unless genre.items.include?(self)
  end
```

Reply:
Thank you for the code hint